tap_hubspot/hubspot.py

```
@@ -4,74 +4,143 @@
-        "forms": ["updatedAt"],
-    }
-
-    def __init__(self, config, tap_stream_id, properties, limit=250):
+
+    def __init__(self, config, limit=250):
         self.SESSION = requests.Session()
         self.limit = limit
         self.access_token = None
-        self.tap_stream_id = tap_stream_id
         self.config = config
         self.refresh_access_token()
-        self.endpoint = self.ENDPOINTS[tap_stream_id]
-        self.offset_value = None
-        self.offset_key = None
-        self.hasmore = True
-        self.PARAMS = {
-            "companies": {"limit": self.limit, "properties": properties,},
-            "contacts": {
-                "showListMemberships": True,
-                "includeVersion": True,
-                "count": self.limit,
-            },
-            "engagements": {"limit": self.limit},
-            "deals": {
-                "count": self.limit,
-                "includeAssociations": False,
-                "properties": properties,
-                "limit": self.limit,
-            },
+
+    def streams(self, tap_stream_id, start_date, end_date, properties):
+        if tap_stream_id == "companies":
+            yield from self.get_companies(properties)
+        elif tap_stream_id == "contacts":
+            yield from self.get_contacts()
+        elif tap_stream_id == "engagements":
+            yield from self.get_engagements()
+        elif tap_stream_id == "deal_pipelines":
+            yield from self.get_deal_pipelines()
+        elif tap_stream_id == "deals":
+            yield from self.get_deals(properties)
+        elif tap_stream_id == "email_events":
+            start_date = self.datetime_to_milliseconds(start_date)
+            end_date = self.datetime_to_milliseconds(end_date)
+            yield from self.get_email_events(start_date, end_date)
+        elif tap_stream_id == "forms":
+            yield from self.get_forms()
+        else:
+            return []
```

I would `raise NotImplementedError(f"unknown stream_id: {tap_stream_id}")` here, since it would otherwise fail silently if something changed in the catalog.